app/javascript/packages/document-capture/components/suspense-error-boundary.tsx

```diff
@@ -0,0 +1,52 @@
+import { Component, Suspense } from 'react';
+import type { ReactNode } from 'react';
+import { RetrySubmissionError } from './submission-complete';
+
+interface SuspenseErrorBoundaryProps {
+  // Fallback to show while suspense pending.
+  fallback: NonNullable<ReactNode>|null;
+  // Error callback.
+  onError: (error: Error)=>void;
+  // Error instance caught to allow for acknowledgment of rerender, in order to prevent infinite rerendering.
+  handledError: RetrySubmissionError | undefined;
```

Comment:
when it's optional like this we can shorthand the syntax:
```suggestion
  handledError?: RetrySubmissionError;
```